flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -778,7 +778,7 @@ private static PartitionInfo createFinishedPartitionInfo(
      */
     @Override
     public void fail(Throwable t) {
-        processFail(t, true);
+        processFail(t, true, Collections.emptyMap());
```

Comment:
Would this be enriched once the https://github.com/apache/flink/pull/22506#discussion_r1185081996 is addressed, since `fromSchedulerNg` will be set to `false`?

Reply:
Again this is covered with the next PR: https://github.com/apache/flink/pull/22511/files#diff-f9e390431932be9a4505e581632815b81c1f0520e65cec7614ea4b90c8a52afcR781-R782

But still there are corner cases that are not easily covered, e.g., [triggerTaskFailover](https://github.com/apache/flink/pull/22511/files#diff-c7f8fc6bbf59b935414529c76720dcafce6105e16783b918af6b06a10e247ab2R103-R104)  [UpdatePartitionInfo](https://github.com/apache/flink/pull/22511/files#diff-f9e390431932be9a4505e581632815b81c1f0520e65cec7614ea4b90c8a52afcR1378)

Thoughts?

Reply:
JM side task failures may originate from SplitEnumerator in SourceCoordinator, which is possible to be a user side problem.
Another kind of failures that users may need to handle is slot allocation timeout exceptions, which may be caused by lack of resources. I see that #22511 targets to handle TM disconnect failures, yet not sure whether this kind of failures are covered.